### DISCUSSION TITLE
Question about recell

### DISCUSSION THREAD

**Discussion by lvdbruno** [lpollier/battmon#7] [Q&A] [answered]
Hello Loïc,

First of all, thank you for the incredible work you've done here. Through your guide, I have gone through with a re-cell, but the success, at least as it seems right now, was only partial.

I used Panasonic NCR18500A cells (measured individually at about 2000-2100mAh), and reprogrammed the EEPROM according to those specs. I assembled the battery and charged it with the ERS-210/220 charging station - it charged up to a certain capacity (the charging station takes it up to the 2nd bar, then stays there for a long time, and then displays the 3 bars flashing), and stopped.  I assume that this has to do with some discrepancy between the capacity set in the EEPROM and what the gas gauge is seeing for this first run. Blew the fuse once, but replaced it with a close variant I found on AliExpress (5A version).

The AIBO runs, however, it took a few presses of the power button to actually get it to start the first time. When it didn't start, the battery lock solenoid engaged, immediately disengaged and the robot turned off. after 3 or 4 presses, it turned on. It has been running for the past hour or so, but it shut down with no warning once already. (edit: If the AIBO is in stationary mode it'll be fine, the moment it starts chasing a ball for example, it will shut down)

I assume that this behavior will go away once the battery discharges, but I have a couple of questions for you, if you'd be so kind:

1 - What do you make of the situation above? Have you seen this kind of behavior before, and if so, do you suspect any particular cause?

2 - Regarding charging and discharging the battery with a balance charger after the pack is assembled: Should I connect the charger terminals directly to the cells, or to the battery header's + and - terminals?

Thank you very much for your time

**Comment by lpollier** [comment]
Hi Bruno,

Thank you for your message and the kind words, and apologies for the late reply.

It seems like you've done a commendable job so far. To address the issue you're facing, first check that there is voltage between Pack+ and Pack- when the switch between /EN and Pack- is turned ON, as shown in the wiring diagram. Next, verify that the I2C interface is functioning correctly. Finally, calibrate the battery to ensure it operates normally.

1. To ensure that all pins of the connector are functioning correctly, I recommend building the BattMon 2.0 toolkit. It's inexpensive and provides real-time data on the battery's state, making it easier to diagnose any issues by reading a few registers of the gas gauge. For my AIBO, all the 4 batteries I recelled started on the first press because I used BattMon to check the main state registers and performed an initial calibration: charging fully, then discharging fully, and then charging again.

2. About this calibration step: Once the battery pack has been repaired and its state confirmed to be okay as explained in point 1, you can use a battery discharger by connecting to Pack+ and Pack-. You should not have access to the central voltage of the two cells since the pack is closed at this step. Ensure that the discharge current is constant, ideally between 0.2 to 0.5A, and check that it reaches the minimal voltage range of 6.0 to 6.3V. At this moment, the gas gauge will store the new actual capacity of your pack in the EEPROM. You can then start to recharge and use your battery with AIBO.

I hope this helps! Let me know if you have any further questions.

Best regards,
Loïc

**Comment by lvdbruno** [comment, marked as answer]
Hi Loïc! Thank you very much for your insights. Over the last few days I did some troubleshooting and eventually found what was causing the issue - there was a significant charge imbalance between the two series groups of cells, which was causing low battery life and overall unstable behavior both in use and in charging. I removed the cells, properly balanced the pack and now it is running perfectly well, with one charge lasting between 2,5 to 3 hours. I am also using the BattMon tool for these re-cells and it is indeed quite handy.


If anyone comes across this answer and is facing similar issues, here are some of the possibilities you have for ensuring that the battery is properly balanced and won't cause you any issues. I have re-celled two so far using two different methods (both with NCR18500A cells), one utilizing a balance charger and the other without.

• Without balance charger:

If you have access to a cell charger (I use the LiitoKala Lii-500, easily found on AliExpress, but any cell charger will do), you can charge all 4 cells to maximum capacity individually before assembly. That way, they will be at nearly equal voltages before assembly and the pack will start off balanced. Be _very careful_ in handling fully charged cells, and when assembling the pack take note that you **must** isolate the negative terminals of the cells before you bring them together and solder the wire that puts them in series, otherwise you will short the leftmost 2-cell group and probably have a very bad time.

I assume that fully discharging all cells will most likely work as well while being much safer, but I did not try it as the second method is preferable:

• With a balance charger & balancing lead

If you have a balance charger available, you can put the pack together at any voltage level and balance it afterwards. You need a balance charger and either buy or build a 3-terminal lead with the correct header (easily found on AliExpress as well). For safety, discharge the cells previously. After you build the new pack, you can solder the balancing tap to the cell's negative, middle and positive terminal, connect it and the charger output to the battery, and run it in balance mode. (this is done before you solder the cells back into the BMS board). Your pack will then be correctly balanced. You can then desolder the balance connector and proceed with the assembly as per Loïc´s guide.


Other insights that may be useful:

- The 5 amp variant of the fuse, the SFD-045A, is as of July 2024, still in abundance over at AliExpress and I can confirm that it works perfectly well.

- Desoldering and soldering the fuse is tricky without a hot-air station, but entirely possible and potentially preferable since you don't risk dislodging other components. For desoldering, first remove as much solder as you can from the three contacts using desoldering wick. More flux will also help you here. If you get lucky, the fuse will dislodge after this step, but if it doesn't, you can easily remove its cover using tweezers, place your soldering iron tip directly on any exposed metal within, while attempting to gently shimmy it using tweezers. It should come loose. For soldering the new one, clean the pads on the board, add flux and a very very tiny amount of solder to each one. do the same on the bottom of the new fuse, then, line up the fuse with the pads on the board and heat up the bottom pad first, while placing a little pressure with your tweezers on it. The fuse should now be soldered to the board. You can then go in with your iron and solder, and do the remaining two easily.

- Before you spot weld the groups of cells together, you should make sure that they are properly aligned and spaced in order to fit correctly inside the shell - it is very easy to weld them slightly skewed, which is enough to make the battery thicker, not close properly and have difficulty in fitting in the AIBO. One way this can be handled is by placing the cells inside the batttery's plastic shell and, while they are resting in their correct positions, secure them in place using tape (preferably Kapton or any other tape that does not leave residue behind). Then, do the first welds with the batteries secured like this, then you can remove the 2-cell group and be sure that they are correctly aligned and spaced. If you do this, the pack will close perfectly well.

- Always take the time to properly insulate all exposed contacts on the cells. This is paramount, since the battery will be constantly moving and shaking along with the AIBO and you don't want your insulation to come loose over time. It is a good idea to put some kapton beneath all of the tabs that will hang out on the side of the cells as well, but, be mindful of how much extra thickness you are adding to the pack, or you may run into assembly issues later.


Good luck, and Loïc, if you see any glaring issues in the above methods, feel free to correct me!

- Bruno

**Reply by lpollier** [reply]
Hi Bruno,

Thanks for sharing your detailed steps to fix your battery cell imbalance issue. It’s great to hear that you were able to resolve the issue effectively.

Indeed, if the cells do not have similar voltage, this may indicate that they are not balanced. In my case, I first checked the real capacity of each of the 4 cells with my B6V2 charger in "Discharge mode," which then allowed me to recharge them in "Storage mode" up to the nominal voltage close to 3.7V, i.e., at their 50% capacity. By the way, the cells are supposed to be stored and shipped at this 50%-60% capacity.

I agree that fully charging the cells before soldering them together seems quite risky, so I would recommend discharging them to the same voltage if it is not possible to balance them with a decent charger.

As for replacing the fuse, I note your choice of the SFD-045A variant. I opted for the SFD-047A with a slightly higher max cut-off current to more closely match the original fuse specification of 6.5A.

Overall, your knowledge and procedures are extremely valuable to anyone undertaking a battery re-celling process. If you have any other tips or updates, please share.

Sincerely,
Loïc

**Comment by GHthekiwi** [comment]
Loïc and Bruno

Are either of you able to provide me with some comments on the tread that I have posted?
I would appreciate soe guiance very much

Thank you
